assets/src/components/map.tsx

```diff
@@ -312,25 +283,88 @@ const Autocenterer = ({
   return <></>
 }
 
-const Map = (props: Props): ReactElement<HTMLDivElement> => {
+const useAutoCenterState = () => {
+  const [shouldAutoCenter, setShouldAutoCenter] = useState<boolean>(true)
+  const isAutoCentering: MutableRefObject<boolean> = useRef(false)
+
+  return {
+    shouldAutoCenter,
+    setShouldAutoCenter,
+    isAutoCentering,
+  }
+}
+
+interface AutoCenterMapOnProps {
+  latLngs: LatLng[]
+  isAutoCentering: MutableRefObject<boolean>
+  shouldAutoCenter: boolean
+  setShouldAutoCenter: Dispatch<SetStateAction<boolean>>
+}
+
+const AutoCenterMapOn = (props: AutoCenterMapOnProps) => {
+  const { shouldAutoCenter, setShouldAutoCenter, isAutoCentering, latLngs } =
+    props
```

Comment:
**suggestion (non-blocking):** could you destructure the `props` in the function argument on the line above instead?

Reply:
[`5c1bd47` (#1890)](https://github.com/mbta/skate/pull/1890/commits/5c1bd476541682377de75c1dd05397a9bf21a80e)